client/allocrunner/taskrunner/template/template_test.go

```diff
@@ -1452,6 +1454,250 @@ func TestTaskTemplateManager_Config_VaultNamespace_TaskOverride(t *testing.T) {
 	assert.Equal(overriddenNS, *ctconf.Vault.Namespace, "Vault Namespace Value")
 }
 
+// TestTaskTemplateManager_Escapes asserts that when sandboxing is enabled
+// interpolated paths are not incorrectly treated as escaping the alloc dir.
+func TestTaskTemplateManager_Escapes(t *testing.T) {
```

Comment:
There's an existing test at https://github.com/hashicorp/nomad/pull/9383/files#diff-f40e85b67f576552c2d39989b55c983267bf19524185cc5f60d4716be0f71359R364 that covers using interpolation to escape. This test setup is nicer so maybe we could move those test cases into this one?

Reply:
I would prefer to leave that test be for now. It's definitely uglier, but it's a bit more comprehensive than this minimal unit test. There's so many state permutations to test in templates I'm pretty nervous reworking that test might miss some subtle state (eg _actually executing the templates_ and meta var interpolation are unique to that test).